api/v1beta1/cloudstackmachine_types.go

```diff
@@ -43,6 +43,9 @@ type CloudStackMachineSpec struct {
 	// CloudStack template to use.
 	Template CloudStackResourceIdentifier `json:"template"`
 
+	// CloudStack disk offering to use.
+	DiskOffering CloudStackResourceDiskOffering `json:"diskOffering,omitempty"`
```

Comment:
should this be a list, in case a customer wants to add multiple disk offerings?

Reply:
According to https://cloudstack.apache.org/api/apidocs-4.14/apis/deployVirtualMachine.html
only one diskofferingId is supported when deploy a VM in cloudstack.

That being said, above api has an alternative parameter `datadiskofferinglist` which can provider an list of disk template to disk offering id map. But `datadiskofferinglist` is designed to a special template which supports multiple disk OVA file)

I did not find any info that qcow2 in cloudstack support multiple data disk, - can confirm with cloudstack experts later.

In the future, if multiple data disk template is supported, the proper places to configure `datadiskofferinglist` should be under templateOffering section since it is template specific and data disk could not be customized.